src/poetry/inspection/info.py

```diff
@@ -266,6 +266,11 @@ def _from_distribution(
 
         :param dist: The distribution instance to parse information from.
         """
+        if dist.metadata_version not in pkginfo.distribution.HEADER_ATTRS:
+            # This check can be replaced once upstream implements strict parsing
+            # https://bugs.launchpad.net/pkginfo/+bug/2058697
+            raise ValueError("Unknown metadata version")
```

Comment:
I think it would be nice to include the metadata version number here: https://github.com/python-poetry/poetry/pull/9226.